web/postgres_typemap.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+func toSpannerTypePostgres(srcType string, spType string, mods []int64) (ddl.Type, []internal.SchemaIssue) {
```

Comment:
This code generalizes toSpannerType from the postgres package: essentially the code there is the same as the default cases in this code.

One question: it seems this code drops the "maxExpectedMods" sanity check -- was that intended.

Also, at some point we should figure out how to avoid the code duplication here and have the postgres package call into this code -- is it sufficient for toSpannerType to just call toSpannerTypePostgres with spType set to the empty string? And maybe this code gets moved into the postgres module so we keep all of the postgres functionality together (and we avoid the Postgres suffix in the function name we have here).

Happy to defer that refactor for the moment and add a TODO for now.

Reply:
Yeah Nevin, i also thought about replacing original toSpannerType with toSpannerTypePostgres, but as that is the core function of conversion and might disturb the command line functionality, i chose not to change it and create a separate function instead for the moment. i'll add a TODO to tackle this later.

The reason why maxExpectedMods is dropped is because it is adding Unexpected statements on top of the default conversion, so the conv.Stats.Unexpected actually becomes inconsistent when user changes type of any column, because it contains the Stats from default conversion as well as manual type changes.

Reply:
Yes, let's add a TODO and defer for the moment.

On the issue of maxExpectedMods: so the main concern here is duplicate reporting of issues? In other words, that we'll report them multiples times if we do type conversion multiple types. I don't see that as a big issue because these issues should rarely if ever arise -- they are essentially an assertion about behaviour of the parser. So the count doesn't matter too much -- it's whether they appear at all.

Having said all of that, I'm wondering if it is time to just remove them from a code-simplicity point of view. The reason I put them in initially was because I was making lots of assumptions when I wrote the first version of HarbourBridge and I hadn't use the pg_query before. However, I haven't seen any of these over the last six months or more. I think they've probably outlive their usefulness -- what do you think?

the absence/presence of an issue is more important that the count.

Reply:
Makes sense to remove for code simplicity, unexpected condition in maxExpectedMods is very rare to arise.